core-api/src/main/java/com/optimizely/ab/config/parser/ConditionJacksonDeserializer.java

```diff
@@ -48,20 +48,37 @@ public ConditionJacksonDeserializer() {
     @Override
     public Condition deserialize(JsonParser parser, DeserializationContext context) throws IOException {
         JsonNode node = parser.getCodec().readTree(parser);
-
-        String conditionsJson = node.textValue();
-        JsonNode conditionsTree = objectMapper.readTree(conditionsJson);
+        JsonNode conditionsTree = node;
+        if (node.isTextual()) {
+            String conditionsJson = node.textValue();
+            conditionsTree = objectMapper.readTree(conditionsJson);
+        }
         Condition conditions = ConditionJacksonDeserializer.parseConditions(objectMapper, conditionsTree);
 
         return conditions;
     }
 
     protected static Condition parseConditions(ObjectMapper objectMapper, JsonNode conditionNode) throws JsonProcessingException {
+
+        if (conditionNode.size() == 0) {
```

Comment:
Does this identify empty arrays, or empty objects, or both? The former (which could map to an OrCondition with no children) should evaluate to true while the latter (which should map to a UserAttribute) should evaluate to null.

Reply:
this would be for empty array only

Reply:
I do not want to add back in parsing as string after the last PR where you told me that audienceConditions would never be a string.  I'm sticking to that.  Sorry.

Reply:
Okay, if things stay like this, I will need to track this stuff and figure out how to account for SDK differences during datafile build (both right now and in the future). I’ll approve for now but we should discuss this class of SDK differences with the team